### PR TITLE
Updates for README

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [ ] All code has been formatted with [`prettier`](https://prettier.io/)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Updates for the dojo 5.0.0 changes in the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -375,12 +385,14 @@ Generates a fully-functional service worker that is activated on startup, comple
...
 ```
 
-#### `build-time-render`: object
+#### `build-time-render`(BTR): object
 
 Renders the application to HTML during the build and in-lines the critical CSS. This allows the application to effectively render static HTML pages and provide some advantages of SSR (server side rendering) such as performance, SEO etc without the complexities of running a server to support full SSR.
 
  * root (required) : The `id` of the root DOM node that application `merge` onto.
- * paths (optional): An array of hash routes to render the application for during the build, for more complex routes an object can be provided with a basic "matcher" (regular expression) that is used to match against the applications route on page load. *Only supports hash routing.*
+ * paths (optional): An array of hash routes to render the application for during the build, for more complex routes an object can be provided with a basic "matcher" (regular expression) that is used to match against the applications route on page load.
```
Comment (agubler) [inline README.md]:
```suggestion
 * paths (optional): An array of routes to render the application for during the build, for more complex routes an object can be provided with a basic "matcher" (regular expression) that is used to match against the applications route on page load.
```

File: README.md [hunk 2]
```diff
@@ -45,10 +45,16 @@ The built application files are written to the `output/{dist/dev}` directory. Th
...
 Note: `dist` is the default mode and so can be run without any arguments, `dojo build app`.
 
-#### Dist Mode
+#### Dist Mode (default)
 
 The `dist` mode creates a production-ready build.
 
+##### Bundle Analyzer
+
+When building your application in `dist` mode, the build process will generate a webpack bundle analyzer that can be used to inspect the content of your applications bundles. The webpack bundle analyzer is outputted to the `output/info/analyzer` directory of your project. To view the analyzer open the `index.html` file contained in this directory.
```
Comment (dylans) [inline README.md]:
```suggestion
When building your application in `dist` mode, the build process will generate a webpack bundle analyzer that can be used to inspect the content of your application's bundles. The webpack bundle analyzer is outputted to the `output/info/analyzer` directory of your project. To view the analyzer, open the `index.html` file contained in this directory.
```

File: README.md [hunk 3]
```diff
@@ -61,6 +67,14 @@ The `unit` mode creates bundles that can be used to run the unit tests of the ap
 
 The `functional` mode creates bundles that can be used to run the functional tests of the application.
 
+### Polyfills
+
+The build command conditionally loads polyfills from `@dojo/framework/shim` based on your applications usage and the users browsers capabilities.
```
Comment (dylans) [inline README.md]:
```suggestion
The build command conditionally loads polyfills from `@dojo/framework/shim` based on your application's usage and the user's browser capabilities.
```

File: README.md [hunk 4]
```diff
@@ -61,6 +67,14 @@ The `unit` mode creates bundles that can be used to run the unit tests of the ap
...
+
+#### Legacy Browser Support
+
+By default, the build will support the last two versions of the latest browsers. To support IE 11 run the build with the `--legacy` (`-l`) flag.
```
Comment (dylans) [inline README.md]:
```suggestion
By default, the build will support the last two versions of the latest browsers. To support IE 11, run the build with the `--legacy` (`-l`) flag.
```

File: README.md [hunk 5]
```diff
@@ -375,12 +385,14 @@ Generates a fully-functional service worker that is activated on startup, comple
...
 ```
 
-#### `build-time-render`: object
+#### `build-time-render`(BTR): object
 
 Renders the application to HTML during the build and in-lines the critical CSS. This allows the application to effectively render static HTML pages and provide some advantages of SSR (server side rendering) such as performance, SEO etc without the complexities of running a server to support full SSR.
 
  * root (required) : The `id` of the root DOM node that application `merge` onto.
- * paths (optional): An array of hash routes to render the application for during the build, for more complex routes an object can be provided with a basic "matcher" (regular expression) that is used to match against the applications route on page load. *Only supports hash routing.*
+ * paths (optional): An array of routes to render the application for during the build, for more complex routes an object can be provided with a basic "matcher" (regular expression) that is used to match against the applications route on page load.
```
Comment (dylans) [inline README.md]:
```suggestion
 * paths (optional): An array of routes for rendering the application during the build; for more complex routes an object can be provided with a basic "matcher" (regular expression) that gets used to match against the application's route on page load.
```

File: README.md [hunk 6]
```diff
@@ -375,12 +385,14 @@ Generates a fully-functional service worker that is activated on startup, comple
...
+
+ Build time rendering supports applications that use either the `@dojo/framework/routing/history/HashHistory` and `@dojo/framework/routing/history/StateHistory` history managers. If your application uses the `HashHistory` ensure that all `paths` are prefixed with a `#` character.
```
Comment (dylans) [inline README.md]:
```suggestion
 Build time rendering supports applications that use either the `@dojo/framework/routing/history/HashHistory` or `@dojo/framework/routing/history/StateHistory` history managers. If your application uses the `HashHistory`, ensure that all `paths` are prefixed with a `#` character.
```

File: README.md [hunk 7]
```diff
@@ -407,7 +421,12 @@ if (!has('build-time-render')) {
 }
 ```
 
-**Note:** The application needs to use the `merge` API from `Projector`
+**Note:** The `index.html` of your application needs to contain a DOM node with the `id` specified as the `root` in the configuration. This DOM node needs to be used when mounting the renderer for application:
```
Comment (dylans) [inline README.md]:
```suggestion
**Note:** The `index.html` of your application needs to contain a DOM node with the `id` specified as the `root` in the configuration. This DOM node needs to be used when mounting the `renderer` for application:
```

**Comment by agubler** [comment]
@dylans thanks, updated.

**Pull request merged**